akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala

```diff
@@ -314,8 +314,8 @@ object Credentials {
         new Credentials.Provided(token) {
           def verify(secret: String, hasher: String => String): Boolean = secret secure_== hasher(token)
         }
-      case Some(GenericHttpCredentials(scheme, token, params)) =>
-        throw new UnsupportedOperationException("cannot verify generic HTTP credentials")
+      case Some(c) =>
+        throw new UnsupportedOperationException(s"Credentials does not support $c.")
```

Comment:
Printing `c` might leak secrets to the exception message. Let's use `c.scheme` instead.

```suggestion
        throw new UnsupportedOperationException(s"Credentials does not support scheme '${c.scheme}'.")
```